src/validator.cpp

```diff
@@ -952,63 +962,131 @@ void Validator::ValidatorImpl::validateMathMLElements(const XmlNodePtr &node, co
     }
 }
 
-void Validator::ValidatorImpl::validateConnections(const ModelPtr &model)
+/**
+ * @brief Test to see if the given variables are reachable from their parent components.
+ *
+ * Determine if the variables parents are reachable from one another.  That is they either
+ * have a parent/child relationship or a sibling relationship in the models component
+ * hierarchy.
+ *
+ * Both variables must have a valid parent.
+ *
+ * @param variable1 The first variable.
+ * @param variable2 The second variable.
+ *
+ * @return True if the parents of the given variables are reachable in the model component
+ * hierarchy.
+ */
+bool reachableEquivalence(const VariablePtr &variable1, const VariablePtr &variable2)
+{
+    bool reachable = false;
+    auto parent1 = variable1->parent();
+    auto parent2 = variable2->parent();
+
+    if (isEntityChildOf(parent1, parent2)) {
+        reachable = true;
+    } else if (isEntityChildOf(parent2, parent1)) {
+        reachable = true;
+    } else if (areEntitiesSiblings(parent1, parent2)) {
+        reachable = true;
+    }
+
+    return reachable;
+}
```

Comment:
I find this kind of code unnecessarily complicated to read. I would simply have:

```
bool reachableEquivalence(const VariablePtr &variable1, const VariablePtr &variable2)
{
    auto parent1 = variable1->parent();
    auto parent2 = variable2->parent();

    return isEntityChildOf(parent1, parent2)
           || isEntityChildOf(parent2, parent1)
           || areEntitiesSiblings(parent1, parent2);
}
```